docker/docker-compose.development.yml

```diff
@@ -1,52 +1,69 @@
 version: '3.5'
 
-services:
+x-networked: &networked
```

Comment:
References:
[Compose file version 3 extension fields](https://docs.docker.com/compose/compose-file/#extension-fields)
[Don't Repeat Yourself with Anchors, Aliases and Extensions in Docker Compose Files](https://medium.com/@kinghuang/docker-compose-anchors-aliases-extensions-a1e4105d70bd)

The main reason for the change in syntax is simply to try to avoid repetition (DRY) as we would in C#.

The change from the array format format to dictionary format for declaring environment variables was because I was unable to get the extension fields to work with the former (I'm not sure it's supported).

While I was conscious that this differs to our existing docker-compose definitions and therefore introduces some tech debt (to make the other definitions consistent), I thought it worthwhile due to the benefits of reduced duplication – the problems of which are well-known.

(As it's YAML, the syntax is fairly easy to pick up too.)